src/resolve/resolved-dns-zone.c

```diff
@@ -519,7 +519,10 @@ void dns_zone_item_notify(DnsZoneItem *i) {
                         we_lost = true;
                 } else {
                         assert(i->probe_transaction->received);
-                        we_lost = memcmp(&i->probe_transaction->received->sender, &i->probe_transaction->received->destination, FAMILY_ADDRESS_SIZE(i->probe_transaction->received->family)) < 0;
+                        we_lost = !dns_resource_key_is_dnssd_ptr(i->rr->key) &&
+                                (memcmp(&i->probe_transaction->received->sender,
```

Comment:
hmm, i think we should probably have a utility can in_addr_cmp() or so, defined in in-addr-util.c, next to in_addr_equal() for this